src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -49,6 +51,35 @@ class NotifyActionPlanService(
   }
 }
 
+@Service
+class NotifyAppointmentService(
+  @Value("\${notify.templates.appointment-not-attended}") private val appointmentNotAttendedTemplateID: String,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.appointment-not-attended}") private val interventionsUIAppointmentNotAttendedLocation: String,
+  private val emailSender: EmailSender,
+  private val hmppsAuthService: HMPPSAuthService,
+) : ApplicationListener<AppointmentEvent>, NotifyService {
+
+  @AsyncEventExceptionHandling
+  override fun onApplicationEvent(event: AppointmentEvent) {
+    when (event.type) {
+      AppointmentEventType.NO_ATTENDANCE -> {
+        val userDetails = hmppsAuthService.getUserDetail(event.appointment.actionPlan.referral.createdBy)
+        val location = generateResourceUrl(interventionsUIBaseURL, interventionsUIAppointmentNotAttendedLocation, event.appointment.id)
+        emailSender.sendEmail(
+          appointmentNotAttendedTemplateID,
+          userDetails.email,
+          mapOf(
+            "ppFirstName" to userDetails.firstName,
+            "referralNumber" to event.appointment.actionPlan.referral.referenceNumber!!,
```

Comment:
we should probably update this var name in the template so it's clear it's expecting the reference number. 'referralNumber' is ambiguous - it could be the referral id too.